### ISSUE TITLE
A bug in mapping network structure?

### ISSUE THREAD

**Issue by YuDeng** [lucidrains/pi-GAN-pytorch#7]
Hi, thanks for sharing the implementation of pi-gan.

I quickly go through the network part and find that there might be a bug in the mapping network:
https://github.com/lucidrains/pi-GAN-pytorch/blob/0067bffdcea65feb552ee29f2c94dc59b3a545f2/pi_gan_pytorch/pi_gan_pytorch.py#L130

In my understanding, the gamma and beta for different SIREN block should be different. In the original styleGAN implementation, the gamma and beta for each AdaIN block are obtained via different fc layers. However in the implementation here, seems that gamma and beta are the same for all layers. 

The original pi-gan paper does not describe this part clearly but only says that gamma and beta comes from the mapping network. I wonder if this different w.r.t the stylegan is a bug or not.

**Comment by ventusff** [comment]
I believe it is claimed in section 3.1 in the original pi-gan paper that the gammas and betas should be different for different SIREN blocks.

They use the i-th subscripts for different layers, thus the `gamma_i` and `beta_i` should also be different for different layers, otherwise the notation would just be `gamma` and `beta`.

![image](https://user-images.githubusercontent.com/25529198/110748786-32cfce80-827b-11eb-90f9-aac76621210e.png)

**Comment by YuDeng** [comment]
Yes, this is also what I expected. Then the implementation in this repo seems to be wrong.

**Comment by primecai** [comment]
I'm wondering if anyone attepted to fix this issue and how's the performance?

Many thanks.
